File: 2058-ConcatenationOfArray/2058-ConcatenationOfArray.java
// Last updated: 5/20/2026, 11:35:57 PM
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] ans = new int[2 * nums.length];
        for(int i =0; i < nums.length; i++){
            ans[i] = ans[i+ nums.length] = nums[i];
        }
        return ans;
    }
}

File: 2058-ConcatenationOfArray/2058-ConcatenationOfArray.cpp
// Last updated: 5/20/2026, 11:35:55 PM
#include <algorithm>
#include <vector>

using std::vector;

class Solution {
    public:
        vector<int> getConcatenation(vector<int>& nums) {
            vector<int> n;
            n.reserve(2 * nums.size());
            n.insert(n.end(), nums.begin(), nums.end());
            n.insert(n.end(), nums.begin(), nums.end());

            return n;
        }
};



